Remove unused stream factory

kelunik opened the pull request:
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | yes
| License         | MIT

#### What's in this PR?

Remove the unused stream factory.

#### Why?

Because it's unused.

kelunik committed:
Remove unused stream factory

Nyholm reviewed (approved):
Good. Just to make sure to add it in the Changelog.

kelunik committed:
Restore useless parameter and emit deprecation notice if passed

This ensures we don't add another parameter which might be a BC break compared to v0.1.0 then.

kelunik committed:
Make deprecation notice opt-in

Nyholm commented:
There is no need to deprecate it. Just remove it and we release 0.2.0. But the change log should state that we removed it.

joelwurtz commented on src/Client.php:
```diff
@@ -19,13 +18,23 @@ class Client implements HttpClient
+    /**
+     * @param Artax\Client    $client          HTTP client implementation.
+     * @param ResponseFactory $responseFactory Response factory to use or `null` to attempt auto-discovery.
+     * @param StreamFactory   $streamFactory   This parameter will be ignored and removed in the next major version.
```
I think you can remove it directly withtout waiting for a next major and tag the next release a 0.2.0, it will be compliant with semver and as it's a fresh library it should be fine.

kelunik replied:
Yes, but that way we can tag `v0.1.1` and nobody even has to update their `composer.json`.

kelunik commented:
We can do that as well, don't really care.

kelunik commented:
@Nyholm I'd just tag v0.1.1 for now, but if you really want to tag v0.2.0, I can change the PR, too.

Nyholm commented:
Thank you for this PR.

Nyholm commented:
Feel free to tag 0.1.1.

kelunik commented:
Tagged.
